### DISCUSSION TITLE
How to customize UI?

### DISCUSSION THREAD

**Discussion by bulkinav** [LunatiqueCoder/react-native-media-console#32] [Q&A] [answered]
Hello,

Thanks for your job. I tried it and I had a questions about customize UI.

For example, now the '**Title**' display not at the top of the screen to the right of the "Back" icon (although for me this position is more familiar). Next, '**time remaining**' also like to place the time on the right.

Perhaps you need to think about support templates for UI, because the default template may not suit everyone (even though he looks good).

**Comment by LunatiqueCoder** [comment, marked as answer]
Hello @bulkinav 

Currently, you can only customise the UI by using the [currently implemented APIs](https://github.com/criszz77/react-native-media-console#-api). 

In the near future, we plan to make it even more customisable, like being able to provide your own icons and styles for elements, maybe even callbacks, but maintaining different templates I think it's too much work.

You can always use [patch-package](https://www.npmjs.com/package/patch-package) to alter your node modules however you want. Or you can just copy paste the source code into your project and customise it precisely however you want, although, you will basically take the whole maintenance burden on your shoulders, instead of having the React Native community taking care of the video player for you.

**Reply by inthroxify** [reply]
Rather than patching packages, if someone went to the trouble of passing style overrides down to the child elements, are you accepting of PRs for that?

**Reply by LunatiqueCoder** [reply]
@inthroxify Yes, we are accepting PRs. However, I would suggest you to start from this branch: https://github.com/LunatiqueCoder/react-native-media-console/tree/expo-app-tv-example

It will be easier to contribute if you start from this branch which we'll merge soon enough.
